### PR TITLE
Fix wrong API requests when `trailingSlash` configuration is true

### PR DESCRIPTION
**Pull request by peterpalau**
Closes: #2294

### What are the changes and their implications?

This PR adds the implementation to consider the `trailingSlash` configuration in the API requests from either await direct function calls inside `useEffects` or `useQuery`.

## Checklist

- [x] Added `trailingSlash` to the blitz runtime dat
- [x] Updated `getApiUrlFromResolverFilePath` to add trailing slash if `getBlitzRuntimeData().trailingSlash` is true.

### CODE REVIEW

File: packages/core/src/rpc-client.ts [hunk 1]
```diff
@@ -157,7 +158,10 @@ executeRpcCall.warm = (apiUrl: string) => {
 }
 
 const getApiUrlFromResolverFilePath = (resolverFilePath: string) =>
-  resolverFilePath.replace(/^app\/_resolvers/, "/api")
+  resolverFilePath.replace(
+    /^app\/_resolvers/,
+    "/api".concat(getBlitzRuntimeData().trailingSlash ? "/" : ""),
+  )
```
Comment (flybayer) [inline packages/core/src/rpc-client.ts:164]: You need to contact the slash on end of resolverFilePath, not `api`. Current code will result in url like `/api//queries/getUser`

**Review by flybayer — CHANGES_REQUESTED** [review]
Thanks!!

Please add an integration test. You can do that by copying `test/integration/queries/` to `test/integration/trailing-slash/` and adding the trailing slash config to that.

**Comment by cashmagnet** [comment]
👋 are things happening here - I don't want to bug anyone for this, just want to know if I should pick it up if needed 🥰

**Comment by peterpalau** [comment]
Hi @cashmagnet

>  are things happening here - I don't want to bug anyone for this, just want to know if I should pick it up if needed

I will be finishing this soon. Thank you.

**Comment by peterpalau** [comment]
I'm sorry, things went wrong here. I'm going to discard this PR.

**Comment by cashmagnet** [comment]
I'm a little confused, I'm really sorry - I just don't want to steal your thunder ⚡ - does this mean you're gonna make another, or just want to be rid of this one and someone else should do it 😄

**Comment by flybayer** [comment]
We have another PR opened for this: 2392

**Pull request closed without merging**